src/crimson/os/futurized_store.h

```diff
@@ -105,8 +137,16 @@ class FuturizedStore {
     CollectionRef c,
     const ghobject_t& oid) = 0;
 
+  virtual seastar::future<std::tuple<std::vector<ghobject_t>, ghobject_t>> list_objects(
+    CollectionRef c,
+    const ghobject_t& start,
+    const ghobject_t& end,
+    uint64_t limit) const = 0;
+
   virtual seastar::future<CollectionRef> create_new_collection(const coll_t& cid) = 0;
+
   virtual seastar::future<CollectionRef> open_collection(const coll_t& cid) = 0;
+
   using coll_core_t = std::pair<coll_t, core_id_t>;
   virtual seastar::future<std::vector<coll_core_t>> list_collections() = 0;
```

Comment:
`list_collections()` is supposed to be a global interface across shards, move it to `FuturizedStore`.

It's only user is `PGShardManager::load_pgs()` called in `OSD::start()`, which is also a part of OSD global initialization.

Let's adjust it to `PGShardManager::load_pgs(FuturizedStore&)` to call this from `FuturizedStore` instead.